sklearn/neighbors/tests/test_kde.py

```diff
@@ -164,8 +171,12 @@ def test_kde_sample_weights():
         n_samples_test = size_test // d
         test_points = rng.rand(n_samples_test, d)
         for algorithm in ["auto", "ball_tree", "kd_tree"]:
-            for metric in ["euclidean", "minkowski", "manhattan", "chebyshev"]:
-                if algorithm != "kd_tree" or metric in KDTree.valid_metrics:
+            for metric in [
+                "euclidean", "minkowski", "manhattan", "chebyshev", "arccos"
+            ]:
+                if ((metric != "arccos" or d != 1) and
+                    (algorithm != "kd_tree" or metric in KDTree.valid_metrics)
+                ):
```

Comment:
Is it possible to parametrize `test_kde_sample_weights` on `metric` and `metric` using the `pytest.mark.parametrize` decorator?

Reply:
Looks to me like that should be possible, yes. You wrote "`metric` and `metric`", but I suppose you meant `algorithm` and `metric`? Shall I parametrize `d` as well, while I'm at it?

Reply:
Yes, that's a typo from me: I meant `algorithm` and `metric`, which can be parametrised with `d` like you propose.

You can also use `pytest.skip` for the combinations of the parameters which aren't meeting the conditions of the test L117 to line 179.